### ISSUE TITLE
About patches_center_locations.shp

### ISSUE THREAD

**Issue by Eliaukyxw** [remicres/sr4rs#54]
Hi,@remicres 
  When pre-training, I had two problems: 1. Can the resolution of HR image and LR image vary by a factor of 5 by changing the patchsize? 2. Is the _patches_center_locations.shp_ file only contain HR image's patches centers? Or also need LR image's patches centers?

**Comment by remicres** [comment]
Hi @Eliaukyxw ,

1. The images resolutions must vary by a factor of `2^n`. You can resample the LR or the HR image depending on the output resolution you want to reach.

2. The *patches_center_locations.shp* contains (LR, HR) patches centers. For each LR patch you must have a target HR patch.

**Comment by Eliaukyxw** [comment]
Got it! Thank you for your prompt reply！I will try the resampling method.

**Comment by remicres** [comment]
Hi @Eliaukyxw 

- The environment variable to change the number of sources is misspelled. It is `OTB_TF_NSOURCES` (not `...NBSOURCES`)
- Your `fid` field has no value. It should be integer type.

**Comment by Eliaukyxw** [comment]
Thank you for your prompt reply！I finally succeeded in creating the patches images！And I'm going to continue with the experiment.😀

**Comment by Eliaukyxw** [comment]
Hi, @remicres 
I'm sorry to bother you again. Is there something wrong with my code?
![4](https://user-images.githubusercontent.com/117817033/207288545-0b9f6967-7ad0-4773-9752-6d7cc78d0abe.png)
![5](https://user-images.githubusercontent.com/117817033/207287629-797990fc-fd72-4c32-8f87-41e299922547.png)
I downloaded the _lr_patches.tif_  and _hr_patches.tif_  to see if they match each other and found that they didn't match:
![6](https://user-images.githubusercontent.com/117817033/207289825-51e21fc0-c1da-457b-bae6-785ea2d9b605.png)
Will it affect the training of the model？And how can I solve this problem?

**Comment by remicres** [comment]
Don't worry at this point the patches have no geographical information. It won't affect the training.
When you open them in QGIS for instance, they are displayed using some pixel spacing information taken from metadata, but the origin in still `[0, 0]` that is why they might look shifted a bit

**Comment by Eliaukyxw** [comment]
Oh I see! Great Remicres! Thanks again, we couldn’t have pulled this off without you.

**Comment by Eliaukyxw** [comment]
Hi, @remicres 
Sorry to bother you again. When my training is interrupted, how do I use checkpoints to continue training?
![8](https://user-images.githubusercontent.com/117817033/208384639-90dda446-ed27-48cc-9946-c73a7af56532.png)
How should the code be written？Hope to get your timely guidance.

**Comment by Eliaukyxw** [comment]
Hi, @remicres 
Thank you for a series of work, with your help I successfully completed the experiment.
Through experiments, I found that OTBTF docker can be used to make LR/HR patches very conveniently. I have a guess: Under this framework, can we use other super resolution networks besides GAN network for training? If so, how can I modify the code file to make it work?

**Comment by remicres** [comment]
Hi @Eliaukyxw ,

Yes of course you can get rid of GAN losses and remove the discriminator. 
You can modify the generator in `networks.py` and develop your own model. Just keep in mind that is must preserve the outputs to inputs resolutions factors (defined in `constants.py`) at the different scales.

**Comment by Eliaukyxw** [comment]
OK, really appreciate your advice! I will have a try!

**Comment by Eliaukyxw** [comment]
Hi, @remicres 
I'm terribly sorry I have so many questions.😂
In `constants.py` , If the LR images to HR images is 8 times larger, factors can be changed to [1,2,4,8], right?

**Comment by remicres** [comment]
Yes.
You also have to adapt the networks so that the tensors dimensions are correct.

**Comment by Eliaukyxw** [comment]
Hi again,
Please forgive my ignorance, because I am new to this field.
I don't know how to modify the code to accommodate 8x magnification.Could you please tell me the detailed suggestions for modification？More specially is this part of the discriminator code in networks.py going to change？
`    with tf.compat.v1.variable_scope(scope, reuse=tf.compat.v1.AUTO_REUSE):
        with tf.compat.v1.variable_scope("res_4x"):
            net = _combine(hr_images[1], newdim=dim, name="from_input")
            net = conv_lrelu(net, dim, 3, 1, "conv1")
            net = conv_lrelu(net, dim, 3, 1, "conv2")
            net = conv_lrelu(net, dim, 3, 1, "conv3")
            net = _conv_downsample(net, dim, 3, "conv_down")

        with tf.compat.v1.variable_scope("res_2x"):
            net = _combine(hr_images[2], newdim=dim, name="from_input", z=net)
            dim *= 2
            net = conv_lrelu(net, dim, 3, 1, "conv1")
            net = conv_lrelu(net, dim, 3, 1, "conv2")
            net = conv_lrelu(net, dim, 3, 1, "conv3")
            net = _conv_downsample(net, dim, 3, "conv_down")

        with tf.compat.v1.variable_scope("res_1x"):
            net = _combine(hr_images[4], newdim=dim, name="from_input", z=net)
            dim *= 2
            net = conv_lrelu(net, dim, 3, 1, "conv")
            net = _conv_downsample(net, dim, 3, "conv_down")

        with tf.compat.v1.variable_scope("bn"):
            dim *= 2
            net = conv_lrelu(net, dim, 3, 1, "conv1")
            net = _conv_downsample(net, dim, 3, "conv_down1")
            net = minibatch_stddev_layer(net)

            # dense
            dim *= 2
            net = conv_lrelu(net, dim, 1, 1, "dense1")
            net = conv(net, 1, 1, 1, "dense2")
            net = tf.reduce_mean(net, axis=[1, 2])

            return net`
And this part of the generator code?
`    with tf.compat.v1.variable_scope(scope, reuse=tf.compat.v1.AUTO_REUSE):
        with tf.compat.v1.variable_scope("encoder"):
            net = lrelu(conv(lr_image, dim, 9, 1, "conv1_9x9"))
            conv1 = net
            for i in range(nresblocks):
                net = _residule_block(net, dim=dim, name="ResBlock{}".format(i))

        with tf.compat.v1.variable_scope("res_1x"):
            net = conv(net, dim, 3, 1, "conv1")
            net = pixel_norm(net)
            net += conv1
            _make_output(net, factor=4)

        with tf.compat.v1.variable_scope("res_2x"):
            net = conv_upsample(net, 4 * dim, 3, "conv_upsample")
            net = conv_bn(net, 4 * dim, 3, "conv1")
            net = conv_bn(net, 4 * dim, 3, "conv2")
            net = conv_bn(net, 4 * dim, 5, "conv3")
            _make_output(net, factor=2)

        with tf.compat.v1.variable_scope("res_4x"):
            net = conv_upsample(net, 4 * dim, 3, "conv_upsample")
            net = conv_bn(net, 4 * dim, 3, "conv1")
            net = conv_bn(net, 4 * dim, 3, "conv2")
            net = conv_bn(net, 4 * dim, 9, "conv3")
            _make_output(net, factor=1)

        return hr_images`
I don't know if I understand correctly? If so, how should it be modified？

**Comment by Eliaukyxw** [comment]
Greetings! @remicres 
Please forgive my ignorance, because I am new to this field.
I want to know if my tif image has only one or two channels, how do I modify the network to make it work?
Looking forward to you early reply!

**Comment by Eliaukyxw** [comment]
Greetings! @remicres 
Please forgive my ignorance, because I am new to this field.
I want to know if my tif image has only one or two channels, how do I modify the network to make it work?
Like the following pictures, an error occurred when input Channel was 1:

Looking forward to you early reply!

**Comment by Eliaukyxw** [comment]
Greetings! @remicres 
Please forgive my ignorance, because I am new to this field.
I want to know if my tif image has only one or two channels, how do I modify the network to make it work?
Like the following pictures, an error occurred when input Channel was 1:
![10](https://user-images.githubusercontent.com/117817033/209622849-7a4df058-b0f1-4ac5-90fb-7753e8f07344.png)
![11](https://user-images.githubusercontent.com/117817033/209622864-ef5ab70d-3bfe-412f-8beb-ee06b8b8a258.png)
Looking forward to you early reply!
